spark-connect/client/src/test/scala-spark-master/io/delta/connect/tables/RemoteSparkSession.scala

```diff
@@ -97,10 +98,14 @@ trait RemoteSparkSession extends BeforeAndAfterAll { self: Suite =>
   override def beforeAll(): Unit = {
     super.beforeAll()
     server
-    // TODO: Instead of sleeping for a fixed time, which is a bit brittle,
-    // we should repeatedly check when the server is ready.
-    Thread.sleep(10000)
-    spark = SparkSession.builder().remote(s"sc://localhost:$serverPort").build()
+
+    val sparkConnectClient = SparkConnectClient
+      .builder()
+      .userId("test")
+      .port(serverPort)
+      .retryPolicy(RetryPolicy.defaultPolicy())
+
+    spark = SparkSession.builder().client(sparkConnectClient.build()).create()
```

Comment:
`build` is deprecated - https://github.com/apache/spark/blob/3edc9c23a723a92c5a951cea0436529de65c640a/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala#L1015